index.js

```diff
@@ -38,110 +28,54 @@ const defaultOptions = {
 
 module.exports = function prettyFactory (options) {
   const opts = Object.assign({}, defaultOptions, options)
-  const EOL = opts.crlf ? '\r\n' : '\n'
-  const IDENT = '    '
   const messageKey = opts.messageKey
   const timestampKey = opts.timestampKey
-  const errorLikeObjectKeys = opts.errorLikeObjectKeys
-  const errorProps = opts.errorProps.split(',')
-  const ignoreKeys = opts.ignore ? new Set(opts.ignore.split(',')) : undefined
 
   const colorizer = colors(opts.colorize)
-  const search = opts.search
-
-  return pretty
 
-  function pretty (inputData) {
-    let log
-    if (!isObject(inputData)) {
-      const parsed = jsonParser(inputData)
-      log = parsed.value
-      if (parsed.err) {
-        // pass through
-        return inputData + EOL
-      }
-    } else {
-      log = inputData
-    }
-
-    // Short-circuit for spec allowed primitive values.
-    if ([null, true, false].includes(log) || Number.isFinite(log)) {
-      return `${log}\n`
-    }
-
-    if (search && !jmespath.search(log, search)) {
-      return
-    }
-
-    if (ignoreKeys) {
-      log = Object.keys(log)
-        .filter(key => !ignoreKeys.has(key))
-        .reduce((res, key) => {
-          res[key] = log[key]
-          return res
-        }, {})
-    }
+  const context = {
+    opts,
+    EOL: opts.crlf ? '\r\n' : '\n',
+    IDENT: '    '
+  }
 
-    const prettifiedLevel = prettifyLevel({ log, colorizer })
-    const prettifiedMessage = prettifyMessage({ log, messageKey, colorizer })
-    const prettifiedMetadata = prettifyMetadata({ log })
-    const prettifiedTime = prettifyTime({ log, translateFormat: opts.translateTime, timestampKey })
+  if (opts.logParsers) {
+    logParsers.push(...opts.logParsers)
```

Comment:
_Sorry to jump in into an interesting PR._

This line would modify the global `logParsers` exported from the `require('./lib/log-parsers')` line, so adding a new instance of `pino` with `prettyPrint` options would use the previously modified (pushed to) `logParsers` too, which is a conflict to me.

You should create local copy of `logParsers`, not modify the global ones.

Reply:
Good catch.

Reply:
Definitely a good catch! I've applied your suggestion.